src/transformers/models/whisper/modeling_whisper.py

```diff
@@ -1593,27 +1617,48 @@ def generate(
                     raise ValueError(
                         f"The `{generation_config.task}`task is not supported. The task should be one of `{TASK_IDS}`"
                     )
-            else:
+            elif hasattr(generation_config, "task_to_id"):
```

Comment:
Thanks @amyeroberts, comments addressed

Only callout I have is that I updated this line to handle the case where an english-only model is being used and the `model.config.forced_decoder_ids` and `generation_config.forced_decoder_ids` are both `None`, since the `generation_config` doesn't have `task_to_id` in that case